test/json_unittest.cc

```diff
@@ -21,7 +28,10 @@ void GuardJsonException(std::function<void()> test) {
 TEST(TrivialParseTest, Null) {
   GuardJsonException([](){
     json::value v = json::Parser::FromString("null");
+    EXPECT_TRUE(v->Is<json::Null>());
```

Comment:
That's separate. You can test that all of the enum values are distinct (which, BTW, we can assert statically in the source — I'll make that change later), but then there's the behavior of the `Is` template... It would be good to confirm that, e.g., `v->Is<json::Number>()` is false for a `json::Boolean`.

Reply:
Done.